### PR TITLE
Create Mongo index for simulation time

### PR DESCRIPTION
**Pull request by thalassemia**
Our default collection indices are out of date. This PR removes unused indices and implements the correct index for simulation time, which should improve performance for queries targeting specific time points or ranges.

-----------------------------------------------------------------------

By creating this pull request, I agree to the Contributor License
Agreement, which is available in `CLA.md` at the top level of this
repository.

### CODE REVIEW

File: vivarium/core/emitter.py [hunk 1]
```diff
@@ -34,15 +34,11 @@
 MONGO_DOCUMENT_LIMIT = 1e7
 
 HISTORY_INDEXES = [
-    'time',
-    'type',
-    'simulation_id',
+    'data.time',
```
Comment (eagmon) [inline vivarium/core/emitter.py:37]: Interesting that `time` is no longer an index. Did you test that `data.time` lets you retrieve the nested field?

**Review by eagmon — APPROVED** [review]
just one question before you merge.

File: vivarium/core/emitter.py (hunk 1 above)
Comment (thalassemia) [reply vivarium/core/emitter.py:37]: Just tested it on my local machine and it works great. Here is the [relevant documentation](https://www.mongodb.com/docs/manual/core/index-single/#create-an-index-on-an-embedded-field).

Comment (eagmon) [reply vivarium/core/emitter.py:37]: LGTM

**Pull request merged**